error compacting TSM files: tsm file closed

version 0.13.0, branch 0.13, commit e57fb88a051ee40fd9277094345fbd47bb4783ce
while loading got error
```
[tsm1] 2016/07/09 19:45:03 beginning level 2 compaction of group 0, 2 TSM files
[tsm1] 2016/07/09 19:45:03 compacting level 2 group (0) /var/lib/influxdb/data/noc/d90/163/000000082-000000002.tsm (#0)
[tsm1] 2016/07/09 19:45:03 compacting level 2 group (0) /var/lib/influxdb/data/noc/d90/163/000000084-000000002.tsm (#1)
[tsm1] 2016/07/09 19:45:03 beginning level 3 compaction of group 0, 4 TSM files
[tsm1] 2016/07/09 19:45:03 compacting level 3 group (0) /var/lib/influxdb/data/noc/d90/163/000000004-000000003.tsm (#0)
[tsm1] 2016/07/09 19:45:03 compacting level 3 group (0) /var/lib/influxdb/data/noc/d90/163/000000008-000000003.tsm (#1)
[tsm1] 2016/07/09 19:45:03 compacting level 3 group (0) /var/lib/influxdb/data/noc/d90/163/000000012-000000003.tsm (#2)
[tsm1] 2016/07/09 19:45:03 compacting level 3 group (0) /var/lib/influxdb/data/noc/d90/163/000000016-000000003.tsm (#3)
[tsm1] 2016/07/09 19:45:16 error compacting TSM files: tsm file closed
[tsm1] 2016/07/09 19:47:30 error compacting TSM files: tsm file closed
```
it looks like database load had stopped. 
no further activity in log for at least 15 minutes and top shows huge cpu load on influxd process

so after 20 minutes in cpu load and no activity in log i got 
```
panic: runtime error: index out of range

goroutine 27 [running]:
panic(0xb5b240, 0xc82000e070)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/influxdb/models.scanMeasurement(0xc82029b6c8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/models/points.go:381 +0x19f
github.com/influxdata/influxdb/models.ParseKey(0x0, 0x0, 0x0, 0x0, 0x1, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/models/points.go:144 +0xae
github.com/influxdata/influxdb/tsdb.MeasurementFromSeriesKey(0x0, 0x0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/meta.go:1834 +0x37
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).addToIndexFromKey(0xc820292000, 0xa3, 0x0, 0x0, 0x2, 0xc8201f0f00, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:314 +0x8e
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).LoadMetadataIndex.func1(0x0, 0x0, 0x1, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:219 +0xc3
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*FileStore).WalkKeys(0xc820230100, 0xc82029ba48, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/file_store.go:233 +0x186
github.com/influxdata/influxdb/tsdb/engine/tsm1.(*Engine).LoadMetadataIndex(0xc820292000, 0xa3, 0xc8201f0f00, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/engine/tsm1/engine.go:213 +0xd7
github.com/influxdata/influxdb/tsdb.(*Shard).Open.func1(0xc82022c7e0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/shard.go:171 +0x2d5
github.com/influxdata/influxdb/tsdb.(*Shard).Open(0xc82022c7e0, 0x0, 0x0)
	/root/go/src/github.com/influxdata/influxdb/tsdb/shard.go:145 +0x3d
github.com/influxdata/influxdb/tsdb.(*Store).loadShards.func1(0xc820154578, 0xc8201e01e0, 0xc8201590e0, 0xc8201f0f00, 0xc8201f5f57, 0x3, 0xc8201fe29b, 0x3, 0xc8201db90f, 0x3)
	/root/go/src/github.com/influxdata/influxdb/tsdb/store.go:177 +0x6ec
created by github.com/influxdata/influxdb/tsdb.(*Store).loadShards
	/root/go/src/github.com/influxdata/influxdb/tsdb/store.go:185 +0xa2e
```
thats looks familiar.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -108,6 +108,7 @@ With this release the systemd configuration files for InfluxDB will use the syst
 - [#7081](https://github.com/influxdata/influxdb/issues/7081): Hardcode auto generated RP names to autogen
 - [#7088](https://github.com/influxdata/influxdb/pull/7088): Fix UDP pointsRx being incremented twice.
 - [#7080](https://github.com/influxdata/influxdb/pull/7080): Ensure IDs can't clash when managing Continuous Queries.
+- [#6990](https://github.com/influxdata/influxdb/issues/6990): Fix panic parsing empty key
 
 ## v0.13.0 [2016-05-12]
 

--- a/models/points.go
+++ b/models/points.go
@@ -401,7 +401,7 @@ func scanMeasurement(buf []byte, i int) (int, int, error) {
 	// Check first byte of measurement, anything except a comma is fine.
 	// It can't be a space, since whitespace is stripped prior to this
 	// function call.
-	if buf[i] == ',' {
+	if i >= len(buf) || buf[i] == ',' {
 		return -1, i, fmt.Errorf("missing measurement")
 	}
 

--- a/models/points_test.go
+++ b/models/points_test.go
@@ -1864,3 +1864,9 @@ func TestNewPointsRejectsMaxKey(t *testing.T) {
 		t.Fatalf("parse point with max key. got: nil, expected: error")
 	}
 }
+
+func TestParseKeyEmpty(t *testing.T) {
+	if _, _, err := models.ParseKey(""); err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+}

